dumpling/export/sql.go

```diff
@@ -763,7 +763,17 @@ func ShowMasterStatus(db *sql.Conn) ([]string, error) {
 		}
 		return rows.Scan(addr...)
 	}
-	const showMasterStatusQuery = "SHOW MASTER STATUS"
+
+	// MySQL 8.4.0 and newer: SHOW BINARY LOG STATUS
+	// TiDB, MariaDB, Old MySQL: SHOW MASTER STATUS
+	showMasterStatusQuery := "SHOW MASTER STATUS"
+	if serverInfo.ServerVersion != nil {
+		if !serverInfo.ServerVersion.LessThan(*minNewTerminologyMySQL) &&
```

Comment:
nit: swap the order, check type is MySQL first

Reply:
9cf43306f3712954d0b8785a807e629e8d8a2507